### PR TITLE
CORTX-32773 - Use master config to generate workload files

### PR DESCRIPTION
**Pull request by ketanarlulkar**
## Problem Statement

-   Using master configuration file to generate S3 workloads

## Design

-   For Bug, Describe the fix here.
-   For new test addition post the link for design

## Coding

   Checklist for Author

-   [x]  Coding conventions are followed and code is formatted (e.g. using pycharm in-built formatter)

## Testing

  Checklist for Author

-   [x]  New/Affected tests are executed
-   [x]  Attach test execution logs

## Review 

  Checklist for Author

-   [x]  JIRA number/GitHub Issue added to PR
-   [x]  PR is self reviewed
-   [x]  Jira and state/status is updated and JIRA is updated with PR link
-   [x]  Check if the description is clear and explained

## Documentation

  Checklist for Author

-   [ ]  Changes done to ReadMe

### CODE REVIEW

**Comment by ketanarlulkar** [comment]
Execution logs: [CORTX-32774|comment-2399075](https://jts.seagate.com/browse/CORTX-32774?focusedCommentId=2399075&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-2399075)

**Review by priyankaborawake — APPROVED** [review]

File: src/commons/yaml_parser.py [hunk 1]
```diff
@@ -32,6 +32,36 @@
...
+        if not existing_params or (set(required) - set(existing_params) != set()):
+            raise AssertionError(f"Minimum required parameters are missing {required} for {test}")
+        tool = config['tool']
+        operation = config['operation']
+        required_params = list(master_cfg[tool][operation].keys()) + required
+        # Check for unknown parameters
+        for param in existing_params:
+            if param not in required_params:
+                raise AssertionError(f"Wrong parameter {param} in {test} test.")
+        to_be_added = required_params - existing_params
```
Comment (DhobaleVishal) [inline src/commons/yaml_parser.py:58]: Log added missing parameters.

File: workload/s3/s3api/multipart_partcopy_random.yaml [hunk 2]
```diff
@@ -139,7 +139,7 @@ test_11:
     min_runtime: 5d
     sessions: 30
     tool: s3api
-    operation: multipart_partcopy
+    operation: multipart_partcopy_random
```
Comment (DhobaleVishal) [inline workload/s3/s3api/multipart_partcopy_random.yaml:142]: Make sure function mapping updated for this changes.

**Review by DhobaleVishal — CHANGES_REQUESTED** [review]

File: src/commons/yaml_parser.py (hunk 1 above)
Comment (ketanarlulkar) [reply src/commons/yaml_parser.py:58]: Done

File: workload/s3/s3api/multipart_partcopy_random.yaml (hunk 2 above)
Comment (ketanarlulkar) [reply workload/s3/s3api/multipart_partcopy_random.yaml:142]: Done

**Review by DhobaleVishal — APPROVED** [review]
LGTM

**Review by VikasKumarC — APPROVED** [review]

**Pull request merged**